src/server/database/leveldb-adaptor.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const levelup = require('levelup')
+const PapanUtils = require('../utils.js')
+
+class LevelCollectionAdaptor {
+  constructor (db) {
+    this.db = db
+  }
+
+  put (doc) {
+    let db = this.db
+
+    return new Promise((resolve, reject) => {
+      db.put(doc._id, JSON.stringify(doc), (err) => {
```

Comment:
Make string conversion before passing the parameter and try catch around it, so that you can reject the promise immediately with the stringify error.